lib/error-matcher.js

```diff
@@ -68,18 +68,37 @@ export default class ErrorMatcher extends EventEmitter {
 
   _parse() {
     this.currentMatch = [];
-    const self = this;
-    const matchFunction = function (match, i, string, regex) {
-      match.id = 'error-match-' + self.regex.indexOf(regex) + '-' + i;
-      this.push(match);
-    };
-    this.regex.forEach((regex) => {
-      if (typeof regex === 'function') {
-        regex(this.output).forEach(function (match) {
-          this.push(match);
-        }.bind(this.currentMatch));
-      } else {
-        require('xregexp').forEach(this.output, regex, matchFunction.bind(this.currentMatch));
+
+    // first run all functional matches
+    this.functions && this.functions.forEach((f, functionIndex) => {
```

Comment:
since `this.functions` is set in the `set` function I feel we can safely assume this will always be an array